modules/backend/formwidgets/RecordFinder.php

```diff
@@ -290,7 +290,12 @@ public function getDescriptionValue()
             return null;
         }
 
-        return $this->relationModel->{$this->descriptionFrom};
+        // Convert array-like notation to dot notation
+        if (preg_match('/\[(.*?)\]/', $this->relationModel, $matches)) {
+            $this->descriptionFrom = str_replace(['[', ']'], ['.', ''], $this->descriptionFrom);
+        }
+
+        return data_get($this->relationModel, $this->descriptionFrom);
```

Comment:
@jaxwilko @bennothommo any thoughts on assigning the modified value back to the local property instead of using a scoped variable at this point? I'm slightly concerned that this current approach causes getDescriptionValue() to have a secondary effect of modifying the value of $this->descriptionFrom in some cases.